### PR TITLE
Adding Meta Trait: Trait Count

### PR DESCRIPTION
**Pull request by vickygos**
# Adding Trait Count to OpenRarity
We are automatically adding TraitCount as a meta trait upon Collection initialization so that the OpenRarity ranking and scoring algorithm takes into consideration trait count as an attribute to impact ranking **for all collections**. We will not modify open rarity scoring or ranking algorithm directly in any form. 

For more detailed information and analysis, please review this blog post -https://mirror.xyz/openrarity.eth/oNo7AmgXopMCKq95gv_Xe0p5pKQ_qHWKzTV11DpFxgE

# Definition of Trait Count
TraitCount is defined to be the sum of unique traits an asset has with a non-empty value
- We consider empty to be where the normalized attribute value is any of `(“”, “none”, None)`

# Methodology
Upon `Collection` initialization, we will modify the `tokens` input by adding in a new `StringAttribute` to the tokens' metadata with:
- `attribute.name` = `“meta trait: trait_count”`
    - Naming to decrease collision with existing trait
- `attribute.value` = token.trait_count()
- Note 1: We will not remove any existing fields, regardless if naming is similiar to “trait count”, “trait_count” etc. The reason is because probability and rankings will not be impacted as probabilities will be the same.
- Note 2: There’s a small chance we override attributes with “meta trait: trait_count” but don’t believe it’s likely enough to matter.

### CODE REVIEW

**Comment by amamujee** [comment]
I modified the text a little

File: open_rarity/resolver/testset_resolver.py [hunk 1]
```diff
@@ -491,13 +508,17 @@ def _rank_diff(rank1: int | None, rank2: int | None) -> int | None:
...
 
     Parameters
     ----------
-    collection : Collection
-        collection
+    collection_with_metadata : Collection
+    dry_run: bool
+        If set to True, the CSV will not be written to disk but returned as an
+        array of rows
+
```
Comment (theelderbeever) [inline open_rarity/resolver/testset_resolver.py:521]: `dry_run` doesn't quite feel right here... Maybe something like `to_console` and it still formats that csv but just prints to stdout?

**Review by theelderbeever — CHANGES_REQUESTED** [review]
That was a big one... A couple comments with possible additions and nits. Nothing major.

File: open_rarity/models/collection.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 from open_rarity.models.token_standard import TokenStandard
 from open_rarity.models.utils.attribute_utils import normalize_attribute_string
 
+TRAIT_COUNT_ATTRIBUTE_NAME = "meta trait: trait_count"
```
[the remark this answers is not in the document]
Comment (vickygos) [reply open_rarity/models/collection.py:14]: Done: `TRAIT_COUNT_ATTRIBUTE_NAME = "meta_trait:trait_count"`

**Review by grantli-os — APPROVED** [review]

**Review by theelderbeever — APPROVED** [review]

**Pull request merged**